src/compiler/crystal/macros/interpreter.cr

```diff
@@ -512,33 +517,33 @@ module Crystal
     end
 
     def visit(node : TupleLiteral)
-      @last = TupleLiteral.map(node.elements) { |element| accept element }.at(node)
+      @last = TupleLiteral.map(node.elements) { |element| accept element }
```

Comment:
Why remove the location of these new nodes? (here and below)

Reply:
For preventing to report macro location as error location. However, I forgot they are used to report macro interpreter error, so this is wrong.

Reply:
> For preventing to report macro location as error location

sry but I still don't get it, could you give an example?

Reply:
@bew You have this code:

```crystal
macro foo
  {% x = 1 %}
  {{ x }} + "foo"
end

foo
```

And you will get such a error with bad location when evaluated macro node has its location:

```console
$ bin/crystal run foo.cr
Error in foo.cr:6: expanding macro

foo
^~~

in foo.cr:2: no overload matches 'Int32#+' with type String
Overloads are:
 - Int32#+(other : Int8)
 - Int32#+(other : Int16)
 - Int32#+(other : Int32)
 - Int32#+(other : Int64)
 - Int32#+(other : Int128)
 - Int32#+(other : UInt8)
 - Int32#+(other : UInt16)
 - Int32#+(other : UInt32)
 - Int32#+(other : UInt64)
 - Int32#+(other : UInt128)
 - Int32#+(other : Float32)
 - Int32#+(other : Float64)
 - Number#+()

  {% x = 1 %}
    ^

```

I think now it is wrong that evaluated macro node has location.

- - -

@asterite You added locations to evaluated macro node in #5597, and you said:

> In theory all nodes should have a location.

However, what is **theory**? I'm wondering when these locations are used and useful.

I think it is creepy that evaluated macro node has location because it is just a value except passed arguments. And I think these locations are never used.